src/ReplayTimeline.h

```diff
@@ -331,13 +379,16 @@ class ReplayTimeline {
    * of two Marks.
    */
   struct InternalMark {
+    // Construct InternalMark from serialized mark data, for `owner` timeline with deserialized `session`
+    InternalMark(ReplayTimeline* owner, const MarkData& serialized, ReplaySession::shr_ptr session);
+
     InternalMark(ReplayTimeline* owner, ReplaySession& session,
                  const MarkKey& key)
         : owner(owner),
           proto(key),
           ticks_at_event_start(session.ticks_at_start_of_current_event()),
-          checkpoint_refcount(0),
-          singlestep_to_next_mark_no_signal(false) {
+          singlestep_to_next_mark_no_signal(false),
+          checkpoint_refcount(0) {
```

Comment:
Yes, I've moved the increment / decrement of `checkpoint_refcount` behind member methods `inc_refcount` and `dec_refcount` (found [here](https://github.com/rr-debugger/rr/pull/3406/files#diff-af90524669aebfeefc9bda4f3a4d3030bb8e6d6d906519f0c71a6580be5fe3aaR418-R426)) and moved checkpoint_refcount to be a private member. I'm unsure if this is wanted though. 

Anyway, doing this, requires the initialization order to be different otherwise clang/gcc complains.